### PR TITLE
Method to process and load sprite sheets

### PR DESCRIPTION
**Pull request by antrikshmisri**
This PR adds the functionality to process/load sprite sheets.

### CODE REVIEW

**Review by skoudoro — COMMENTED** [review]
Hi @antrikshmisri,

This PR is ready to go. Can you address the small comments below. 

Thanks !

File: fury/io.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import vtk
 import numpy as np
 from PIL import Image
+from tempfile import TemporaryDirectory as InTemporaryDirectory
```
[the remark this answers is not in the document]
Comment (antrikshmisri) [reply fury/io.py]: Done.

File: fury/io.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
...
 from vtk.util import numpy_support
 from fury.utils import set_input
 from urllib.request import urlretrieve
```
[the remark this answers is not in the document]
Comment (antrikshmisri) [reply fury/io.py]: Fixed above as this is referencing the same line.

File: fury/tests/test_io.py [hunk 3]
```diff
@@ -155,3 +157,23 @@ def test_pillow():
             data2 = load_image(fname_path, use_pillow=opt2)
             npt.assert_array_almost_equal(data, data2)
             npt.assert_equal(data.dtype, data2.dtype)
+
+
+def test_load_sprite_sheet():
+    sprite_URL = 'https://i.imgur.com/0yKFTBQ.png'
```
[the remark this answers is not in the document]
Comment (antrikshmisri) [reply fury/tests/test_io.py]: The image is now hosted on one of my github repository.

File: fury/io.py [hunk 4]
```diff
@@ -281,3 +282,51 @@ def save_polydata(polydata, file_name, binary=False, color_array_name=None):
...
+
+    Parameters
+    ----------
+    sheet_path: str
+        Path to the sprite sheet
+    nb_rows: int
+        Number of rows in the sprite sheet
+    nb_cols: int
+        Number of columns in the sprite sheet
+    as_vtktype: bool
```
[the remark this answers is not in the document]
Comment (antrikshmisri) [reply fury/io.py]: Done.

File: fury/io.py [hunk 5]
```diff
@@ -281,3 +282,51 @@ def save_polydata(polydata, file_name, binary=False, color_array_name=None):
...
+        Number of columns in the sprite sheet
+    as_vtktype: bool, optional
+        If True, the output is a vtkImageData
+
+    Returns
+    -------
+    Dict containing the processed sprites.
+    """
+    sprite_dicts = {}
+    sprite_sheet = Image.fromarray(load_image(sheet_path))
```
Comment (skoudoro) [inline fury/io.py:306]: I do not think you need `Image.fromarray`. The NumPy array should be enough for what you need to do. 
it is also easy to crop a NumPy array via slicing or to get the width or height.

File: fury/io.py [hunk 6]
```diff
@@ -281,3 +282,51 @@ def save_polydata(polydata, file_name, binary=False, color_array_name=None):
...
+    width, height = sprite_sheet.size
+
+    sprite_size_x = width // nb_rows
+    sprite_size_y = height // nb_cols
```
Comment (skoudoro) [inline fury/io.py:310]: how do you handle mistakes like `nb_rows > width` which create a floating value?

File: fury/io.py [hunk 7]
```diff
@@ -281,3 +282,51 @@ def save_polydata(polydata, file_name, binary=False, color_array_name=None):
...
+
+    for row in range(nb_rows):
+        for col in range(nb_cols):
+            nxt_row = row + 1
+            nxt_col = col + 1
+
+            box = (row*sprite_size_x, col*sprite_size_y,
+                   nxt_row*sprite_size_x, nxt_col*sprite_size_y)
+
+            sprite_arr = sprite_sheet.crop(box)
```
Comment (skoudoro) [inline fury/io.py:320]: use slicing of numpy array

File: fury/io.py [hunk 8]
```diff
@@ -281,3 +282,51 @@ def save_polydata(polydata, file_name, binary=False, color_array_name=None):
...
+    """
+    sprite_dicts = {}
+    sprite_sheet = Image.fromarray(load_image(sheet_path))
+    width, height = sprite_sheet.size
+
+    sprite_size_x = width // nb_rows
+    sprite_size_y = height // nb_cols
+
+    for row in range(nb_rows):
+        for col in range(nb_cols):
```
Comment (skoudoro) [inline fury/io.py:313]: `for row, col in np.ndindex((nb_cols, nb_rows))` or something similar

File: fury/io.py [hunk 9]
```diff
@@ -281,3 +282,51 @@ def save_polydata(polydata, file_name, binary=False, color_array_name=None):
...
+
+            sprite_arr = sprite_sheet.crop(box)
+            if as_vtktype:
+                with InTemporaryDirectory() as tdir:
+                    tmp_img_path = os.path.join(tdir, f'{row}{col}.png')
+                    save_image(np.asarray(sprite_arr), tmp_img_path,
+                               compression_quality=100)
+
+                    sprite_dicts[f'{row}{col}'] = load_image(tmp_img_path,
+                                                             as_vtktype=True)
```
Comment (skoudoro) [inline fury/io.py:328]: strange to use the disk for this... Can you add a Todo and an issue saying that we need to replace this. 

Tag it as beginner-friendly

File: fury/io.py [hunk 10]
```diff
@@ -281,3 +282,51 @@ def save_polydata(polydata, file_name, binary=False, color_array_name=None):
...
+            else:
+                sprite_dicts[f'{row}{col}'] = np.asarray(sprite_arr)
```
Comment (skoudoro) [inline fury/io.py:330]: use a tuple as a key instead of this string.

**Review by skoudoro — COMMENTED** [review]
Hi @antrikshmisri,

Thank you for the update. See below for some comments.

File: fury/io.py (hunk 6 above)
Comment (antrikshmisri) [reply fury/io.py]: With the latest commit, if a cell size gets less than 1 it is rounded to 1 as there is no point in generating more slices of that small size.

File: fury/io.py (hunk 9 above)
Comment (antrikshmisri) [reply fury/io.py]: I am using the disk to actually then load the saved image as `vtk.vtkImageData` and the same functionality is required in the `render_tex` method. I guess a function called `numpy_to_vtk_image_data` needs to be added to remove the need to write to disk.

**Review by skoudoro — APPROVED** [review]
Hi @antrikshmisri, 

LGTM, thank you for the update! merging.

**Pull request merged**